### PR TITLE
tests: improve 'ignore-running' spread test

### PR DESCRIPTION
**Pull request by MiguelPires**
Adds some comments to the new refresh-app-awareness test for `refresh-mode: ignore-running` and makes some of its checks a bit more strict.

### CODE REVIEW

File: tests/main/refresh-mode-ignore-running/task.yaml [hunk 1]
```diff
@@ -17,14 +17,17 @@ execute: |
 
   trap 'touch stamp' EXIT
 
+  # refresh app awareness prevents the snap from being refreshed while it's running
   test-snapd-refresh.sh -c "while [ ! -e stamp ]; do sleep 1; done" &
-  not snap install --dangerous ./test-snapd-refresh_1_all.snap
+  snap install --dangerous ./test-snapd-refresh_1_all.snap 2>&1 | tr -s "\n" " " | MATCH ".*snap \"test-snapd-refresh\" has running apps \(sh\), pids:.*"
```
Comment (mvo5) [inline tests/main/refresh-mode-ignore-running/task.yaml:22]: I would do this in two lines:
```
not snap install --dangerous ./test-snapd-refresh_1_all.snap 2>error.txt
tr -s "\n" " " error.txt | MATCH ".*snap \"test-snapd-refresh\" has running apps \(sh\), pids:.*"
```
(note that I did not actually test that I did not do typos here)

mainly to ensure that the "snap install" also errors with an non-zero exit code (we test this elsewhere too for a different error case so one might argue it's redundant). And then the second line tests the actually error message.

The shell will by default ignore any failures on the left hand side of the pipe if we would not error in snap instal this new test would not catch that.

**Review by mvo5 — COMMENTED** [review]
Thanks for these improvements, nice to see. One suggestion about the test inline.

**Review by mardy — APPROVED** [review]
LGTM with the split suggested by Michael.

**Review by mvo5 — APPROVED** [review]
Thank you!

**Pull request merged**